src/decoder.js

```diff
@@ -1,59 +1,186 @@
+    return this.methodIDs;
+  }
+  /**
+   * Decode constructor
+   *
+   * @param Object contract creation code
+   * @return {}
+   */
+  decodeConstructor(contractCreationCode) {
+    if (this.methodIDs.constructor.type !== 'constructor') throw new Error(`Expected constructor got${this.methodIDs.constructor.type}`);
+    const abiItem = this.methodIDs.constructor;
+    if (abiItem) {
+      const params = abiItem.inputs.map(item => item.type);
+      const decoded = SolidityCoder.decodeParams(params, contractCreationCode.slice(10));
+      return {
+        name: abiItem.type,
+        params: decoded.map((param, index) => {
+          let parsedParam = param;
+          if (abiItem.inputs[index].type.indexOf('uint') !== -1) {
+            parsedParam = web3.toBigNumber(param).toString();
+          }
+          return {
+            name: abiItem.inputs[index].name,
+            value: parsedParam,
+            type: abiItem.inputs[index].type,
+          };
+        }),
+      };
+    }
+    return null;
+  }
+  /**
+   * Decode transaction input data
+   *
+   * @param Object inputData
+   * @return {}
+   */
+  decodeMethod(inputData) {
+    const errorObject = { name: 'UNDECODED', params: [{ name: 'rawData', value: inputData, type: 'data' }] };
+    if (inputData === '0x') return '';
+    const methodID = inputData.slice(2, 10);
+    const abiItem = this.methodIDs[methodID];
+    if (abiItem) {
+      const params = abiItem.inputs.map(item => item.type);
+      const decoded = SolidityCoder.decodeParams(params, inputData.slice(10));
+      return {
+        name: abiItem.name,
+        params: decoded.map((param, index) => {
+          let parsedParam = param;
+          if (abiItem.inputs[index].type.indexOf('uint') !== -1) {
+            parsedParam = web3.toBigNumber(param).toString();
+          }
+          return {
+            name: abiItem.inputs[index].name,
+            value: parsedParam,
+            type: abiItem.inputs[index].type,
+          };
+        }),
+      };
+    }
+    return errorObject;
+  }
+  /**
+   * Decode transaction logs
+   *
+   * @param Object logs
+   * @return [{}]
+   */
+  decodeLogs(logs) {
+    const errorObject = { name: 'UNDECODED', events: [{ name: 'rawLogs', value: logs, type: 'logs' }] };
+    const decodedLogs = logs.map((logItem) => {
+      const methodID = logItem.topics[0].slice(2);
+      const method = this.methodIDs[methodID];
+      if (method) {
+        const logData = logItem.data;
+        const decodedParams = [];
+        let dataIndex = 0;
+        let topicsIndex = 1;
+
+        const dataTypes = [];
+        method.inputs.map((input) => {
+          if (!input.indexed) {
+            dataTypes.push(input.type);
+          }
+          return dataTypes;
+        });
+        const decodedData = SolidityCoder.decodeParams(dataTypes, logData.slice(2));
+        // Loop topic and data to get the params
+        method.inputs.map((param) => {
+          const decodedP = {
```

Comment:
I suggest to use clear variable name, I don't know what is the meaning of `decodedP`